### ISSUE TITLE
Simple cats IOApp hello world binary not running

### ISSUE THREAD

**Issue by AlterEgo7** [bazelbuild/rules_scala#870]
Hello, I'm trying to evaluate Bazel for building Scala apps and I'm trying to build a simple hello world IOApp, but get a class not found exception. 

I'm using bazel-deps to define my dependencies.

### WORKSPACE: 
```python
workspace(name = "bazel_poc")

rules_scala_version="0f89c210ade8f4320017daf718a61de3c1ac4773"

load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "io_bazel_rules_scala",
    strip_prefix = "rules_scala-%s" % rules_scala_version,
    type = "zip",
    url = "https://github.com/bazelbuild/rules_scala/archive/%s.zip" % rules_scala_version,
)

load("@io_bazel_rules_scala//scala:toolchains.bzl", "scala_register_toolchains")

scala_register_toolchains()

load("@io_bazel_rules_scala//scala:scala.bzl", "scala_repositories")

scala_repositories()

load("//3rdparty:workspace.bzl", "maven_dependencies")

maven_dependencies()

protobuf_version="09745575a923640154bcf307fba8aedff47f240a"
protobuf_version_sha256="416212e14481cff8fd4849b1c1c1200a7f34808a54377e22d7447efdf54ad758"

skylib_version = "0.8.0"
http_archive(
    name = "bazel_skylib",
    type = "tar.gz",
    url = "https://github.com/bazelbuild/bazel-skylib/releases/download/{}/bazel-skylib.{}.tar.gz".format (skylib_version, skylib_version),
    sha256 = "2ef429f5d7ce7111263289644d233707dba35e39696377ebab8b0bc701f7818e",
)
```

### src/main/scala/example/BUILD:
```python
load("@io_bazel_rules_scala//scala:scala.bzl", "scala_library", "scala_binary")

package(default_visibility = ["//visibility:public"])

scala_binary(
  name = "hello",
  srcs = ["Hello.scala"],
  scalacopts = [
    "-encoding", "UTF-8",
    "-feature",
    "-language:existentials",
    "-language:experimental.macros",
    "-language:higherKinds",
    "-language:implicitConversions",
    "-Ypartial-unification",
  ],
  deps = [
    "//3rdparty/jvm/org/typelevel:cats_core",
    "//3rdparty/jvm/org/typelevel:cats_kernel",
    "//3rdparty/jvm/org/typelevel:cats_macros",
    "//3rdparty/jvm/org/typelevel:cats_effect",
  ],
  main_class = "example.Hello",
  unused_dependency_checker_mode = "warn",
)
```

### bazel_deps.yaml:
```yaml
options:
  buildHeader: [ "load(\"@io_bazel_rules_scala//scala:scala_import.bzl\", \"scala_import\")" ]
  languages: [ "scala:2.12.8" ]
  resolverType: "coursier"

dependencies:
  org.typelevel:
    cats:
      lang: scala
      modules: [ "core", "effect", "kernel", "macros" ]
      version: "2.0.0"
```

### Hello.scala:
```scala
package example

import cats.data.Kleisli
import cats.effect._
import cats.implicits._

object Hello extends IOApp {

  val greet = Kleisli[IO, String, Unit] { name =>
    IO { println(s"Hello $name") }
  }

  override def run(args: List[String]) =
    greet.run("John Doe").as(ExitCode.Success)

}
```

bazel-bin/src/main/scala/example/hello output:
```
Exception in thread "main" java.lang.NoClassDefFoundError: cats/effect/IOApp$class
	at example.Hello$.<init>(Hello.scala:7)
	at example.Hello$.<clinit>(Hello.scala)
	at example.Hello.main(Hello.scala)
Caused by: java.lang.ClassNotFoundException: cats.effect.IOApp$class
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 3 more
```

Am I missing some configuration? I feel like it has to do with the fact that both `IOApp` and `App` are traits but that's just a hunch. (Normal class with main method works fine).

## Sidenote:

The following also fails: 

```scala
object Hello extends App {
  println("test")
}
```

Output:
```
Exception in thread "main" java.lang.AbstractMethodError: Method example/Hello$.scala$App$_setter_$executionStart_$eq(J)V is abstract
	at example.Hello$.scala$App$_setter_$executionStart_$eq(Hello.scala)
	at scala.App$class.$init$(App.scala:44)
	at example.Hello$.<init>(Hello.scala:18)
	at example.Hello$.<clinit>(Hello.scala)
	at example.Hello.main(Hello.scala)
```

**Comment by AlterEgo7** [comment]
Apparently this was an issue of scala version mismatch. The fix is to add the scala-2.12 jar shas in `scala_repositories()` call in WORKSPACE.

**Comment by ittaiz** [comment]
👍🏽 Good luck
